### PR TITLE
NK65 Addition

### PR DESCRIPTION
**Pull request by yiancar**
NK65 by novelkeys. This is an upcoming product.

## Description

65% keyboard, Improvement on some Wilba RGB code.
IS31FL3733 driver has been made more modular.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by yiancar** [comment]
@Wilba6582 please take a look:)

**Comment by mechmerlin** [comment]
OMGGG YESSS I HAVE BEEEN WAAAITING

File: keyboards/nk65/config.h [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x8968
+#define PRODUCT_ID      0x4E4B
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Yiancar-Designs
+#define PRODUCT         NK65
+#define DESCRIPTION     A 65 persent, tool free RGB keyboard
```
Comment (noroadsleft) [inline keyboards/nk65/config.h]:
```suggestion
#define DESCRIPTION     A 65-percent, tool-free RGB keyboard
```

File: keyboards/nk65/readme.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+## Instructions
+
+### Build
+
+Make example for this keyboard (after setting up your build environment):
+
+    make nk65:default_via
+
+See [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) then the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
```
Comment (noroadsleft) [inline keyboards/nk65/readme.md]: Could you update this please?

```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/nk65/readme.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+NK65
+=========
+
+[NK65]()
```
Comment (noroadsleft) [inline keyboards/nk65/readme.md]: There should be an image here of some sort, if possible. Something like:

```suggestion
![NK65](http://yiancar-designs.com/wp-content/uploads/2019/05/NK65_top.jpg)
```

Otherwise this line should be deleted IMO.

Comment (yiancar) [reply keyboards/nk65/readme.md]: I left it blank to be added with a nice render when I have one :) I can comment it out right now or leave it as an empty link and add the pic when I got it

**Comment by yiancar** [comment]
> Additional to the below, some of the `.c` and `.h` files have had their permissions changed to executable, and I can't see any reason why that would be necessary.
> 
> Specifically:
> 
>     * drivers/issi/is31fl3733.c
> 
>     * drivers/issi/is31fl3733.h
> 
>     * keyboards/zeal60/rgb_backlight.c
> 
>     * quantum/led_matrix_drivers.c
> 
>     * quantum/rgb_matrix_drivers.c
> 
> 
> all have been `chmod`ed from `644` to `755`.

Done, sorry I was a dummy

File: keyboards/nk65/readme.md (hunk 3 above)
Comment (noroadsleft) [reply keyboards/nk65/readme.md]: Just as long as it doesn't get forgotten. Commenting out might work too, but I don't know what GitHub does if you try to comment out an image.

Comment (yiancar) [reply keyboards/nk65/readme.md]: Lets go as an empty link then if that's OK

Comment (noroadsleft) [reply keyboards/nk65/readme.md]: :thinking: ...

Yeah, that's fine.

File: keyboards/nk65/rules.mk [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+OPT_DEFS =
+
+# Do not put the microcontroller into power saving mode
+# when we get USB suspend event. We want it to keep updating
+# backlight effects.
+OPT_DEFS += -DNO_SUSPEND_POWER_DOWN
+
+# Options to pass to dfu-util when flashing
+DFU_ARGS = -d 0483:df11 -a 0 -s 0x08000000:leave
+DFU_SUFFIX_ARGS = -p DF11 -v 0483
```
Comment (drashna) [inline keyboards/nk65/rules.mk]: ❤️

File: docs/zh-cn/getting_started_introduction.md [hunk 5]
```diff
@@ -1,54 +1,54 @@
...
-样板代码和设置看起来像这样：
-
-```
-#pragma once
-
-// 像下面那样覆盖设置（MY_SETTING指的是你要覆盖的设置项）!
-#undef MY_SETTING
-#define MY_SETTING 4
-```
+# 介绍
```
Comment (noroadsleft) [inline docs/zh-cn/getting_started_introduction.md]: If you could revert the changes to this file so it's not included in the PR, that would be great. (This file has been fixed in master, and will preserve the Git history.)

Comment (yiancar) [reply docs/zh-cn/getting_started_introduction.md]: Yea i figured but mac os does not let me revert it back. It probably has some sort of "fix line ending" feature embedded in the os. Any ideas?

Comment (noroadsleft) [reply docs/zh-cn/getting_started_introduction.md]: Ooh, I don't use macOS.

@mechmerlin @yanfali Any ideas?

Comment (drashna) [reply docs/zh-cn/getting_started_introduction.md]: I think I had to change the autocrlf and eol settings for git, and then reset to head (hard).

Comment (noroadsleft) [reply docs/zh-cn/getting_started_introduction.md]:
```sh
git config --global core.autocrlf false
git config --global core.eol lf
```
